File: management/src/types/analytics.ts
export type AnalyticsPageType = 'all' | 'product' | 'solution' | 'about' | 'contact' | 'other'
export type AnalyticsPeriodPreset = 'previous_month' | 'this_month' | 'this_week' | 'previous_week' | 'this_year'

export interface AnalyticsDailyRow {
  date: string
  visits_pv: number
  visits_uv: number
  product_page_pv: number
  product_page_uv: number
  solution_page_pv: number
  solution_page_uv: number
  about_page_pv: number
  about_page_uv: number
  contact_page_pv: number
  contact_page_uv: number
}

export interface AnalyticsReport {
  start_date: string
  end_date: string
  rows: AnalyticsDailyRow[]
}

export interface PageViewLog {
  id: number
  visitor_id: string
  session_id: string
  page_type: Exclude<AnalyticsPageType, 'all'>
  path: string
  page_title: string | null
  ip_address: string
  referrer: string | null
  user_agent: string | null
  language: string | null
  screen_size: string | null
  is_bot: boolean
  created_at: string
}

export interface PageViewLogList {
  items: PageViewLog[]
  total: number
  page: number
  page_size: number
}

export interface AnalyticsFilters {
  startDate: string
  endDate: string
  pageType?: AnalyticsPageType
  query?: string
  page?: number
  pageSize?: number
}

File: backend/app/schemas/analytics.py
from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, Field


class PageViewCreate(BaseModel):
    visitor_id: str = Field(min_length=8, max_length=80, pattern=r"^[A-Za-z0-9._:-]+$")
    session_id: str = Field(min_length=8, max_length=80, pattern=r"^[A-Za-z0-9._:-]+$")
    path: str = Field(min_length=1, max_length=500)
    page_title: str | None = Field(default=None, max_length=300)
    referrer: str | None = Field(default=None, max_length=1000)
    language: str | None = Field(default=None, max_length=32)
    screen_size: str | None = Field(default=None, max_length=32)


class PageViewAccepted(BaseModel):
    accepted: bool = True


class AnalyticsDailyRow(BaseModel):
    date: date
    visits_pv: int
    visits_uv: int
    product_page_pv: int
    product_page_uv: int
    solution_page_pv: int
    solution_page_uv: int
    about_page_pv: int
    about_page_uv: int
    contact_page_pv: int
    contact_page_uv: int


class AnalyticsReportOut(BaseModel):
    start_date: date
    end_date: date
    rows: list[AnalyticsDailyRow]


class PageViewLogOut(BaseModel):
    id: int
    visitor_id: str
    session_id: str
    page_type: str
    path: str
    page_title: str | None
    ip_address: str
    referrer: str | None
    user_agent: str | None
    language: str | None
    screen_size: str | None
    is_bot: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class PageViewLogListOut(BaseModel):
    items: list[PageViewLogOut]
    total: int
    page: int
    page_size: int


AnalyticsPageType = Literal["all", "product", "solution", "about", "contact", "other"]
